### DISCUSSION TITLE
Only edit on a resource?

### DISCUSSION THREAD

**Discussion by mwkcoding** [filamentphp/filament#598] [Help] [answered]
I've made a resource where I want to disable delete and create for that resource globally. I can't just remove the route and page. That just makes Filament complain about the route not existing.

**Comment by wsamoht** [comment, marked as answer]
Thanks to @danharrin, he gave me this solution in Discord.

To disable create and delete, return `false` from `canCreate` and `canDelete` in your list class.

*Example*
```php
class ListUsers extends ListRecords
{
    public static $resource = UsersResource::class;

    public function canCreate()
    {
        return false;
    }

    public function canDelete()
    {
        return false;
    }
}
```

To disable editing, add `recordActions([])` to your table method in the resource.

*Example*
```php
// UsersResource.php
public static function table(Table $table)
{
    return $table
        ->columns([
            Text::make('name'),
        ])
        ->filters([
            //
        ])->recordActions([]);
}
```

**Reply by dominikgeimer** [reply]
Thanks for sharing! I am sure that this will be useful for so many people.

**Reply by Khant-Nyar** [reply]
can we use now !
